templates/admin/navbar_admin_products.html.twig

```diff
@@ -85,44 +85,54 @@
 </nav>
 
 <script>
-    var limit = document.getElementById("limit");
-    var order = document.getElementById("sortby");
+    const limit = document.getElementById("limit");
+    const order = document.getElementById("sortBy");
+    const name = document.getElementById("myInput");
 
-    if (window.location.href === 'http://localhost:8000/products/') {
-        limit.value = localStorage.getItem('limit');
-        order.value = localStorage.getItem('order');
+    const urlParams = new URLSearchParams(window.location.search);
 
-        localStorage.setItem('limit', '10');
-        localStorage.setItem('order', 'created_at:DESC');
+    if (urlParams.has("limit")) {
+        limit.value = urlParams.get("limit");
+    } else {
+        limit.value = 10;
     }
 
-    limit.value = localStorage.getItem('limit');
-    order.value = localStorage.getItem('order');
+    if (urlParams.has("order")) {
+        order.value = urlParams.get("order");
+    } else {
+        order.value = "created_at:DESC";
+    }
 
-    function changeLimit(value) {
+    if (urlParams.has("name")) {
+        name.value = urlParams.get("name");
+    }
 
+    function changeLimit(value) {
         var urlSearchParams = new URLSearchParams(window.location.search);
-        urlSearchParams.set('limit', value);
-        urlSearchParams.set('page', 1);
+        urlSearchParams.set("limit", value);
+        urlSearchParams.set("page", 1);
         window.location.search = urlSearchParams;
-
-        localStorage.setItem('limit', value);
     }
 
     function changeSort(value) {
-
         var urlSearchParams = new URLSearchParams(window.location.search);
-        urlSearchParams.set('order', value);
+        urlSearchParams.set("order", value);
         window.location.search = urlSearchParams;
-
-        localStorage.setItem('order', value);
     }
 
     function setName() {
         var inputVal = document.getElementById("myInput").value;
-        var urlSearchParams = new URL('http://localhost:8000/admin/products/');
-        urlSearchParams.searchParams.set('name', inputVal);
-        urlSearchParams.searchParams.set('page', 1);
-        window.location.replace(urlSearchParams);
+        var urlSearchParams;
+        if (window.location.origin + window.location.pathname !== "http://localhost:8000/admin/products/") {
```

Comment:
dont use absolute URL - if I will try to run this project on a different machine with a proper url - it will not work

Reply:
Here I'll think later